src/main/java/com/iexec/common/chain/IexecHubAbstractService.java

```diff
@@ -149,13 +171,47 @@ public IexecHubAbstractService(Credentials credentials,
         }
         this.retryDelay = nbBlocksToWaitPerRetry * (int)this.blockTime.toMillis();
         this.maxRetries = maxRetries;
+        this.expectedFinalDeadlineRatio = expectedFinalDeadlineRatio;
 
         String hubAddress = getHubContract().getContractAddress();
         log.info("Abstract IexecHubService initialized (iexec proxy address) " +
                 "[hubAddress:{}]", hubAddress);
         setMaxNbOfPeriodsForConsensus();
     }
 
+    @PostConstruct
+    private void validateChainConfiguration() {
```

Comment:
I would split the validation in 2 parts:
1. A node exists and can reply (see rpc call to getNodeVersion)
2. A business check like you do, shortly verifying we not connected to a wrong smartcontract (what you already do)

About (2.), I'm not very van of the semantic but I don't have good suggestions so far. The question is "Why introducing this particular field?".

Reply:
Maybe we could wrap `expectedFinalDeadlineRatio` into something like a `IexecSmartContractConnectionPolicy` class?

Reply:
Part (1.) might be done in `Web3jService`

Reply:
That looks way better, thanks. New version is online, let me know if it suits you :)